### PR TITLE
Create auth

### PR DESCRIPTION
**Pull request by joaothallis**

### CODE REVIEW

File: lib/name_of_the_day/accounts.ex [hunk 1]
```diff
@@ -0,0 +1,350 @@
+defmodule NameOfTheDay.Accounts do
+  @moduledoc """
+  The Accounts context.
+  """
+
+  import Ecto.Query, warn: false
+  alias NameOfTheDay.Repo
+
+  alias NameOfTheDay.Accounts.{User, UserToken, UserNotifier}
```
Comment (yashin5) [inline lib/name_of_the_day/accounts.ex:9]: disgusting
```suggestion
  alias NameOfTheDay.Accounts.Use
  alias NameOfTheDay.Accounts.UserToken
  alias NameOfTheDay.Accounts. UserNotifier
```

Comment (yashin5) [reply lib/name_of_the_day/accounts.ex:9]: for sure i'll open a PR to change this in the autogenerate

File: lib/name_of_the_day/accounts.ex [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+
+      iex> get_user_by_email_and_password("foo@example.com", "correct_password")
+      %User{}
+
+      iex> get_user_by_email_and_password("foo@example.com", "invalid_password")
+      nil
+
+  """
+  def get_user_by_email_and_password(email, password)
+      when is_binary(email) and is_binary(password) do
```
Comment (yashin5) [inline lib/name_of_the_day/accounts.ex:42]:
```suggestion
  def get_user_by_email_and_password(<<email>>, <<password>>) do
```

File: lib/name_of_the_day/accounts.ex [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+      iex> get_user_by_email("foo@example.com")
+      %User{}
+
+      iex> get_user_by_email("unknown@example.com")
+      nil
+
+  """
+  def get_user_by_email(email) when is_binary(email) do
+    Repo.get_by(User, email: email)
+  end
```
Comment (yashin5) [inline lib/name_of_the_day/accounts.ex:27]:
```suggestion
  def get_user_by_email(<<email>>) do
    Repo.get_by(User, email: email)
  end
```

File: lib/name_of_the_day/accounts.ex [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+  ## Examples
+
+      iex> get_user!(123)
+      %User{}
+
+      iex> get_user!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  def get_user!(id), do: Repo.get!(User, id)
```
Comment (yashin5) [inline lib/name_of_the_day/accounts.ex:61]: this autogenerate don't make o @spec lol

**Review by yashin5 — APPROVED** [review]

File: lib/name_of_the_day/accounts.ex (hunk 3 above)
Comment (joaothallis) [reply lib/name_of_the_day/accounts.ex:27]: The `<<>>` syntax verify that the variable is a bitstring this includes binaries, using is_binary accepts only string/bitstring:

```elixir
iex(1)> is_binary <<1::3>>
false
iex(2)> is_bitstring <<1::3>>
true
```

Due to this reason, I will keep `is_binary/1` guard

File: lib/name_of_the_day/accounts.ex (hunk 2 above)
Comment (joaothallis) [reply lib/name_of_the_day/accounts.ex:42]: https://github.com/joaothallis/name-of-the-day/pull/1/files#r702459334

File: lib/name_of_the_day/accounts.ex (hunk 1 above)
Comment (joaothallis) [reply lib/name_of_the_day/accounts.ex:9]: I understand your point related to it but I like this way. With find references to functions and modules ElixirLS feature I found that solve the problem of search for places using a module in a project

**Pull request merged**